hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -2957,6 +2957,19 @@ public boolean processReport(final DatanodeID nodeID,
     return !node.hasStaleStorages();
   }
 
+  // Remove the lease when we have received block reports for all storages for a particular DN.
```

Comment:
Remove the lease when we have received block reports for all storages for a particular DN.

->

Remove the DN lease only when we have received block reports for all storages for a particular DN.